src/libstd/io/cursor.rs

```diff
@@ -289,6 +295,16 @@ impl Write for Cursor<Vec<u8>> {
     fn flush(&mut self) -> io::Result<()> { Ok(()) }
 }
 
+#[stable(feature = "cursor_box_slice", since = "1.4.0")]
```

Comment:
There could be a doc comment here explaining the semantics. It *should* be visible on the type's rustdoc page in the current implementation (so on Cursor's page).

The vector growing and the box merely filling is something that we probably want to put into the documentation somehow.